flink-baseline/flink-canary/src/main/java/com/ness/flink/canary/pipeline/function/HealthCheckFunction.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021-2023 Ness Digital Engineering
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ness.flink.canary.pipeline.function;
+
+import com.ness.flink.canary.pipeline.domain.KafkaConfigs;
+import java.util.Collection;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.streaming.api.functions.ProcessFunction;
+import org.apache.flink.util.Collector;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.DescribeClusterOptions;
+import org.apache.kafka.common.Node;
+
+@Slf4j
+public class HealthCheckFunction extends ProcessFunction<KafkaConfigs, String>{
+    private static final long serialVersionUID = 1L;
+    @Override
+    public void processElement(KafkaConfigs value, ProcessFunction<KafkaConfigs, String>.Context ctx, Collector<String> out) throws Exception {
+        String result;
+
+        Properties props = new Properties();
+        String bootStrapServers = value.getBootStrapServers();
+        props.put("bootstrap.servers", bootStrapServers);
```

Comment:
I'm pretty sure that these parameters are not enough to use this solution with Confluent Cloud or with AWS MSK
For example what about these parameters?
```
 security.protocol: SASL_SSL
 sasl.mechanism: PLAIN
 sasl.jaas.config: "org.apache.kafka.common.security.plain.PlainLoginModule   required username='API_KEY' password='API_PASSWD';"
```

Actually I suggest to use KafkaConsumerProperties as a internal message **or even better create new class**
KafkaAdminProperties which will `extends KafkaProperties`
Why?
Because here you are trying to configure Kafka **AdminClient** (not KafkaConsumer or KafkaProducer).

Reply:
Done